### DISCUSSION TITLE
add uart output support in vprintf function

### DISCUSSION THREAD

**Discussion by yoshinoyatoko** [memtest86plus/memtest86plus#255] [Enhancement]
Hi
Sometimes, user could want to debug source code to trace some problems.
Currently, it seems debug messages could be output in screen.
Maybe we could enhance vprintf implementation, add uart output support.

Thanks

**Comment by yoshinoyatoko** [comment]
Any comments?
So, how to debug source code if user wants to trace source code to find some failed clues?

**Reply by debrouxl** [reply]
The pending #177 can help debugging, but I'm not sure whether it would qualify on your target setup, even after some QEMU configuration changes.
On my side, I have partially rebased the combined patch from #177 after the version number changes; I'll post an updated patch there if you're interested.

**Reply by yoshinoyatoko** [reply]
Hi, debrouxl
Thanks a lot.
I mean if user wants to debug source code on a real motherboard, maybe uart output is a good choice.
